packages/websocket-plugin/src/websocket-handler.ts

```diff
@@ -17,157 +16,156 @@ import {
   WebSocketConnected
 } from './symbols';
 
-@Injectable()
+@Injectable({ providedIn: 'root' })
 export class WebSocketHandler implements OnDestroy {
-  private socket: WebSocketSubject<any> | null = null;
-
-  private config: WebSocketSubjectConfig<any> = {
-    url: this.options.url!,
-    protocol: this.options.protocol,
-    // Default binary type is `blob` for the global `WebSocket`
-    binaryType: this.options.binaryType,
-    serializer: this.options.serializer,
-    deserializer: this.options.deserializer,
-    closeObserver: {
-      next: () => {
-        // ATTENTION!
-        // See https://github.com/ReactiveX/rxjs/blob/master/src/internal/observable/dom/WebSocketSubject.ts#L340
-        // RxJS socket emits `onComplete` event only if `event.wasClean` is truthy
-        // and doesn't complete socket subject if it's falsy
-        this.disconnect();
-      }
-    },
-    openObserver: {
-      next: () => this.store.dispatch(new WebSocketConnected())
-    }
-  };
+  private _socket: WebSocket | null = null;
 
-  private typeKey = this.options.typeKey!;
+  private readonly _socketClosed$ = new Subject<void>();
 
-  private subscription = new Subscription();
+  private readonly _typeKey = this._options.typeKey!;
+
+  private readonly _destroy$ = new Subject<void>();
```

Comment:
```suggestion
  private readonly _destroy$ = new ReplaySubject<void>();
```
It is always better to use a replay subject for the _destroy$ observable in case a subscription happens after the service is destroyed.

Reply:
The subscription happens only in constructor.